test/e2e/suite/addons/addons_test.go

```diff
@@ -0,0 +1,117 @@
+	"github.com/aws/eks-hybrid/test/e2e/suite"
+)
+
+var (
+	filePath    string
+	suiteConfig *suite.SuiteConfiguration
+)
+
+const numberOfNodes = 1
+
+func init() {
+	flag.StringVar(&filePath, "filepath", "", "Path to configuration")
+}
+
+func TestE2E(t *testing.T) {
+	RegisterFailHandler(Fail)
+	RunSpecs(t, "Addon Smoke Test Suite")
+}
+
+var _ = SynchronizedBeforeSuite(
+	func(ctx context.Context) []byte {
+		suiteConfig := suite.BeforeSuiteCredentialSetup(ctx, filePath)
+		test := suite.BeforeVPCTest(ctx, &suiteConfig)
+		credentialProviders := suite.AddClientsToCredentialProviders(suite.CredentialProviders(), test)
+		osList := suite.OSProviderList(credentialProviders)
+
+		// pick 3 random OS/Version/Provider combinations for metricsServer tests worker nodes
+		nodesToCreate := make([]suite.NodeCreate, 0, numberOfNodes)
+
+		rand.Shuffle(len(osList), func(i, j int) {
+			osList[i], osList[j] = osList[j], osList[i]
+		})
+
+		for i := range numberOfNodes {
+			os := osList[i].OS
+			provider := osList[i].Provider
+			nodesToCreate = append(nodesToCreate, suite.NodeCreate{
+				OS:           os,
+				Provider:     provider,
+				InstanceName: test.InstanceName("addon-smoke-test", os, provider),
+				InstanceSize: e2e.Large,
+				NodeName:     fmt.Sprintf("addon-test-node-%s-%s", provider.Name(), os.Name()),
+			})
+		}
+		suite.CreateNodes(ctx, test, nodesToCreate)
+
+		suiteJson, err := yaml.Marshal(suiteConfig)
+		Expect(err).NotTo(HaveOccurred(), "suite config should be marshalled successfully")
+		return suiteJson
+	},
+	// This function runs on all processes, and it receives the data from
+	// the first process (a json serialized struct)
+	// The only thing that we want to do here is unmarshal the data into
+	// a struct that we can make accessible from the tests. We leave the rest
+	// for the per tests setup code.
+	func(ctx context.Context, data []byte) {
+		suiteConfig = suite.BeforeSuiteCredentialUnmarshal(ctx, data)
+	},
+)
+
+var _ = Describe("Hybrid Nodes", func() {
+	When("using peered VPC", func() {
+		var addonEc2Test *suite.AddonEc2Test
+		var createSucceeded bool
+
+		BeforeEach(func(ctx context.Context) {
+			createSucceeded = false
+			addonEc2Test = &suite.AddonEc2Test{PeeredVPCTest: suite.BeforeVPCTest(ctx, suiteConfig)}
+		})
+
+		When("using ec2 instance as hybrid nodes", func() {
+			It("runs node monitoring agent tests", func(ctx context.Context) {
+				nodeMonitoringAgent := addonEc2Test.NewNodeMonitoringAgentTest()
+
+				DeferCleanup(func(ctx context.Context) {
+					report := CurrentSpecReport()
+					if report.State.Is(types.SpecStateFailed) && createSucceeded {
+						err := nodeMonitoringAgent.CollectLogs(ctx)
+						if err != nil {
+							// continue cleanup even if logs collection fails
+							GinkgoWriter.Printf("Failed to collect node monitoring agent logs: %v\n", err)
+						}
+					}
+					Expect(nodeMonitoringAgent.Delete(ctx)).To(Succeed(), "should cleanup node monitoring agent successfully")
+				})
```

Comment:
why don't you divide this in two defers?
one for the Delete and one for the logs
you can leave the delete up here
and define the logs one only after the create is succesful